core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -4818,10 +4818,10 @@ object ZIO extends ZIOCompanionPlatformSpecific with ZIOCompanionVersionSpecific
    * not use this method, use [[ZIO.suspend]].
    */
   def suspendSucceed[R, E, A](zio: => ZIO[R, E, A])(implicit trace: Trace): ZIO[R, E, A] =
-    ZIO.succeed(zio).flatMap(identityFn)
+    ZIO.FlatMap(trace, Exit.unit, (_: Any) => zio)
```

Comment:
As per my other comment, let's use `ZIO.unit` to align with #9393. Also, since it might not be obvious as to why we're using `ZIO.unit` here and we might end up destroying optimizations without realising, let's add a test along these lines:

```scala
    test("ZIO.suspendSucceed is implemented with a ZIO.unit as first") {
      ZIO.suspendSucceed(ZIO.succeed(42)) match {
        case ZIO.FlatMap(_, first, _) => assertTrue(first eq ZIO.unit)
        case _ => assertNever("ZIO.suspendSucceed is not implemented as a FlatMap")
      }
    }
```

Reply:
Done :)